src/python/geoclaw/data.py

```diff
@@ -529,3 +529,70 @@ def write(self, out_file='./multilayer.data', datasource="setrun.py"):
                         description=('(Method for calculating inundation ',
                                      'eigenspace)'))
         self.close_data_file()
+
+
+# Point source data
+class PointSourceData(clawpack.clawutil.data.ClawData):
+    """Data object describing point sources"""
+
+    def __init__(self):
+        """Constructor of PointSourceData class"""
+
+        super(PointSourceData, self).__init__()
+
+        # number of point sources
+        self.add_attribute('n_point_sources', 0)
```

Comment:
Do you need this?  Seems like `len(self.point_sources)` would always be equal to this.

Reply:
It's just for preventing users from providing mismatched data in `setrun.py`. So the program is able to identify mistakes in the provided input data before a simulation starts. It's just like the concept of defensive driving. I'll prefer to keep it as this doesn't cause any serious performance overhead and provides an early-stage debugging layer.

Reply:
So you are trying to save the user from themselves?  In general I could see a user being frustrated by adding another point-source and then forgetting to increment `n_point_sources`.  If a user accidentally left out a point source I would hazard the guess that they would probably realize it especially considering I don't think you are going to assume that you have more than a very small number of point sources?